### ISSUE TITLE
Javascript RAML parser for Java

### ISSUE THREAD

**Issue by bademux** [raml-org/raml-java-parser-2#18]
"The parser makes use of Webpack bundled Javascript RAML parser via the Nashorn scripting engine"
Please tell me it is proof of concept.

Let me rephrase,
Will the official RAML parser for java use Javascript raml parser inside Nashorn?

Thanks

**Comment by sichvoge** [comment]
You don't have to worry. There will be something native coming.

**Comment by bademux** [comment]
thanks!
I waiting for it, cos the quality of swagger-codegen makes me cry like a baby :(

**Comment by bademux** [comment]
Let me ask more questions about new parser:
1. Is there ETA?
2. Do you think about generating parser to JS with GWT's [JS Interop](https://docs.google.com/document/d/10fmlEYIHcyead_4R1S5wKGs1t2I7Fnp_PaNaa7XTEk0/edit)
3. Google`s  [j2objc](https://github.com/google/j2objc) allows you to export it to binary library

**Comment by sichvoge** [comment]
1. No ETA yet; we'd like to open source our development **asap** - that version is not GA though, but help our community to see and test our progress.
2. I don't think so. The JS parser itself will be native.
3. That probably is a good idea. Question is if that makes sense or not? You can translate something into something else quite often, but that usually comes with some downsides. An y experience with that?

**Comment by bademux** [comment]
And I hope the last one:
Will RAML java parser v8 be compatible with v1?

Let me explain  my curiosity. I'm choosing standard (and tooling) for my project.
Native tools (JS for javascript, binary for iOS devs, java for Android) can make them happy (faster builds, better tooling integration) without rewriting tooling. So potentially I'm interested, just need more information about the project.
Thanks!

**Comment by svacas** [comment]
regarding v08 and v1 compatibility, the same parser will be able to parse both spec versions, though it will return objects that implement the interfaces for their respective version

**Comment by bademux** [comment]
Hi,
thanks again for the clarification!
One more question:

How issues for v0.8  will be solved?

Found some architectural (I believe) problems with v0.8 parser [here ](https://github.com/raml-org/raml-java-parser/issues/99) and [here](https://github.com/raml-org/raml-java-parser/issues/100).
Will it done right for 1.0 or legacy implementation will be fixed too ?

**Comment by svacas** [comment]
All the current parser issues will be reviewed on the new parser implementation

**Comment by sichvoge** [comment]
Please use the following [link](https://github.com/raml-org/raml-java-parser/tree/v2) for the alpha version of the native Java parser. This repo will be deprecated soon.
